sdk/containerregistry/container-registry/src/containerRepositoryClient.ts

```diff
@@ -133,10 +137,15 @@ export class ContainerRepositoryClient {
     options: ContainerRegistryClientOptions = {}
   ) {
     this.endpoint = endpointUrl;
+    this.repository = repository;
+    const matches = endpointUrl.match(/:\/\/([a-zA-Z0-9.]+)\/*/);
```

Comment:
Would it be possible to use the WHATWG URL API instead of doing regex matching? I always get a little nervous doing regex matching on user-provided inputs...

Reply:
We don't expose URL platform-agonistic wrapper from core-client. For now I will keep a copy of the code in ACR.

Reply:
Now using URL to parse url and retrieve hostname